src/lib/entities/audio/model/audio-instance.ts

```diff
@@ -1,9 +1,53 @@
-import { writable } from 'svelte/store';
+import { writable, derived } from 'svelte/store';
 
-export const audio = writable<HTMLAudioElement>();
+export const audio = writable<HTMLAudioElement | undefined>(undefined);
+
+/** Tracks the duration of the audio stream that is currently loaded in the `audio` store. */
+export const audioDuration = derived<typeof audio, number>(
+  audio,
+  ($audio, set) => {
+    if ($audio === undefined) {
+      set(NaN);
+      return;
+    }
+
+    function watchDuration(event: Event) {
+      set((event.target as HTMLAudioElement).duration);
+    }
+
+    $audio.addEventListener('durationchange', watchDuration);
+
+    return () => $audio?.removeEventListener('durationchange', watchDuration);
```

Comment:
Yeah, of course, we don't know when this callback will be called. I'm not sure if TypeScript will complain, but it should

Reply:
We know for sure that this callback will not be reached if `$audio` was `undefined`. It also threw me off a little that TS didn't complain, but after playing around with it, I realized that it's static analysis was a lot smarter; if it sees `$audio = undefined;` anywhere in the function, it will not make that assumption and will require `?.`, but since it is in a closure and guards against `undefined` and never assigns it, the `$audio` parameter can never refer to `undefined` inside this callback, even if the original `audio` store became `undefined`.
The same applies in the `audioPosition` derived store as well.